crossbar/twisted/resource.py

```diff
@@ -77,6 +79,244 @@ def render_GET(self, request):
+                fileupload_publish({
+                                   "fileId": fileId,
+                                   "fileName": filename,
+                                   "totalSize": totalSize,
+                                   "status": "FINISH",
+                                   "progress": 1
+                                   })
+            else:
+                # first of more chunks
+                os.makedirs(fileTempDir)
+                chunk = open(chunkName, 'wb')
+                chunk.write(fileContent)
+                chunk.close
+
+                # publish file upload progress to file_progress_URI
+                fileupload_publish({
+                                   "fileId": fileId,
+                                   "fileName": filename,
+                                   "totalSize": totalSize,
+                                   "status": "PROGRESS",
+                                   "progress": chunkSize / float(totalSize)
+                                   })
+
+        else:
+            # intermediate chunk
+            chunk = open(chunkName, 'wb')
+            chunk.write(fileContent)
+            chunk.close
+
+            prog = float(sum(os.path.getsize(os.path.join(fileTempDir, f)) for f in os.listdir(fileTempDir))) / totalSize
+
+            fileupload_publish({
+                               "fileId": fileId,
+                               "fileName": filename,
+                               "totalSize": totalSize,
+                               "status": "PROGRESS",
+                               "progress": prog
+                               })
+
+            if chunkNumber == totalChunks:
+                # last chunk
+                chunk = open(chunkName, 'wb')
+                chunk.write(fileContent)
+                chunk.close
+
+                # Now merge all files into one file and remove the temp files
+                finalFile = open(os.path.join(self._dir, fileId), 'wb')
+
+                for tfileName in os.listdir(fileTempDir):
+                    tfile = open(os.path.join(fileTempDir, tfileName), 'r')
+                    finalFile.write(tfile.read())
+
+                finalFile.close()
+
+                try:
+                    perm = int(self._file_permissions, 8)
+                    os.chmod(finalFileName, perm)
+
+                except Exception as e:
+                    request.setResponseCode(500, "File permissions could not be changed")
+                    self.removeTempDir(fileTempDir)
+                    return ''
+
+                # publish file upload progress to file_progress_URI
+
+                fileupload_publish({
+                                   "fileId": fileId,
+                                   "fileName": filename,
+                                   "totalSize": totalSize,
+                                   "status": "FINISH",
+                                   "progress": 1
+                                   })
+
```

Comment:
Sorry for nit-picking, but I think it would be more-consistent to use the 4-space indenting (you've done this in some places for dicts). This also matches other indenting in crossbar, and IMO is more "normal" for Python. That is, like:

```
fileupload_publish({
    "key": value,
    "foo": bar
})
```

Also, not sure if it *really* matters for published information, but crossbar/autobahn use `under_score` naming, in general, not `camelCase` (Twisted uses camelCase, of course, so there's some of that in the code too :/).

But, for "internal" method calls like `removeTempDir` and variables (e.g. `fileTempDir`), I think `under_score` naming should be preferred to keep things consistent.